serverless/networking/serverless-ossm-tls.adoc

```diff
@@ -0,0 +1,168 @@
+----
+. Sign the request with your CA:
++
+[source,terminal]
+----
+$ openssl x509 -req -days 365 -CA example.com.crt -CAkey example.com.key -set_serial 0 -in custom.example.com.csr -out custom.example.com.crt
+----
+. Check that the certificates appear in your directory:
++
+[source,terminal]
+----
+$ ls -1
+----
++
+.Example output
+[source,terminal]
+----
+custom.example.com.crt
+custom.example.com.csr
+custom.example.com.key
+example.com.crt
+example.com.key
+----
+. Create a secret:
++
+[source,terminal]
+----
+$  oc create -n istio-system secret tls custom.example.com --key=custom.example.com.key --cert=custom.example.com.crt
+----
+. Attach the secret to the Istio ingress gateway, by editing the `ServiceMeshControlPlane` resource.
+.. Edit the `ServiceMeshControlPlane` resource:
++
+[source,terminal]
+----
+$ oc edit -n istio-system ServiceMeshControlPlane <control_plane_name>
+----
+.. Check that the following lines exist in the resource, and if they do not, add them:
++
+[source,yaml]
+----
+istio-ingressgateway:
+  secretVolumes:
+  - mountPath: /custom.example.com
+    name: custom-example-com
+    secretName: custom.example.com
+----
+. Update the Istio ingress gateway to use your secret.
+.. Edit the `default-gateway` resource:
++
+[source,terminal]
+----
+$ oc edit gateway default-gateway
+----
+.. Check that the following lines exist in the resource, and if they do not, add them:
++
+[source,yaml]
+----
+- hosts:
+  - custom-ksvc-domain.example.com
+  port:
+    name: https
+    number: 443
+    protocol: HTTPS
+  tls:
+    mode: SIMPLE
+    privateKey: /custom.example.com/tls.key
+    serverCertificate: /custom.example.com/tls.crt
+----
+. Update the route to use pass-through TLS and `8443` as the `spec.port.targetPort`.
+.. Edit the route:
++
+[source,terminal]
+----
+$ oc edit route -n istio-system hello
+----
+.. Add the following configuration to the route:
++
+[source,yaml]
+----
+spec:
+  host: custom-ksvc-domain.example.com
+  port:
+    targetPort: 8443
+  tls:
+    insecureEdgeTerminationPolicy: None
+    termination: passthrough
+  to:
+    kind: Service
+    name: istio-ingressgateway
+    weight: 100
+  wildcardPolicy: None
+----
+
+.Verification steps
+
+* Access your serverless application by a secure connection that is now trusted by the CA:
++
+[source,terminal]
+----
+$ curl --cacert example.com.crt \
+  --connect-to "custom-ksvc-domain.example.com:443:${<ingress-router-IP-or-hostname>}:443" \
+  https://custom-ksvc-domain.example.com
+----
++
+[NOTE]
+====
+You must substitute your own value for `<ingress-router-IP-or-hostname>`.
+Steps for finding this IP or hostname value may vary depending on your {product-title} deployment.
+
+.Example command to find the ingress IP
+[source,terminal]
+----
+$ oc get svc -n openshift-ingress router-default -o jsonpath='{.status.loadBalancer.ingress[0].ip}'
+----
+====
+// Add link for service mesh docs once they're available
```

Comment:
@cardil does this look better to you now? Any other changes or additional info we need to add?

Reply:
Related issue for SM docs raised by Kenjiro: #26032

Reply:
I'm perfectly fine with giving just this example for a specific provider platform. Users should figure out what is needed in their case.